### DISCUSSION TITLE
How to train rectangular images?

### DISCUSSION THREAD

**Discussion by dlin95123** [ultralytics/ultralytics#5011] [Q&A]
Hello there, I have 640x480 images for training. From the cfg default.yaml file, imgsz can only be set as int for training. Then I set rect as True. 

>imgsz: 640  # (int | list) input images size as int for train and val modes, or list[w,h] for predict and export modes
>rect: True  # (bool) rectangular training if mode='train' or rectangular validation if mode='val'

Does that mean the images will be trained at 640x480 resolution? Thanks.

PS. I have searched for similar discussion but couldn't find a clear answer.

**Comment by glenn-jocher** [comment]
@dlin95123 hello! When you set `rect: True` in the configuration, YOLOv8 will indeed support rectangular training, which means that your 640x480 images can be used without resizing them to squares. The `imgsz` parameter defines the longest side of the training images, so with `imgsz: 640`, the network will dynamically adjust the image size during training to maintain the aspect ratio of your original images. This allows for efficient use of GPU memory and can potentially improve performance on datasets with non-square images. Happy training! 😊🚀
